core/src/main/java/org/infinispan/partitionhandling/impl/PartitionHandlingInterceptor.java

```diff
@@ -143,6 +146,22 @@ private BasicInvocationStage handleDataReadCommand(InvocationContext ctx, DataCo
                // get response.
                throw log.degradedModeKeyUnavailable(dataCommand.getKey());
             } else {
+               // If all owners left and we still haven't received the availability update yet,
+               // we get OutdatedTopologyException from BaseDistributionInterceptor.retrieveFromProperSource
+               if (t instanceof OutdatedTopologyException && performPartitionCheck(rCtx, dataCommand)) {
+                  // Unlike in PartitionHandlingManager.checkRead(), here we ignore the availability status
+                  // and we only fail the operation if _all_ owners have left the cluster.
+                  // TODO Move this to the availability strategy when implementing ISPN-4624
```

Comment:
ISPN-4624 is about how a cache changes its availability mode, not about how individual operations react to a particular availability mode. I'm not sure we want to make that customizable.